docs/config-generator.md

```diff
@@ -0,0 +1,66 @@
+# Writing a generic configuration generator class
+
+It is possible to write a class which programmatically generates your nyc configuration.
+In this case a `toNYCRC()` method should be provided which returns the actual configuration.
+
+A simple example which allows you to append the default exclude list instead of replacing
+it:
+
+```js
+'use strict';
+
+const {defaultExclude} = require('test-exclude');
+
+class ConfigGenerator {
```

Comment:
maybe we should publish this as a preset package? `@istanbuljs/base-config-generator` which people could then extend? is that overkill?